### DISCUSSION TITLE
[New Feature] Red Alert 2 like Spy

### DISCUSSION THREAD

**Discussion by unknown-men** [Vinifera-Developers/Vinifera#1272] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [x] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

https://modenc.renegadeprojects.com/Agent

https://modenc.renegadeprojects.com/The_Infiltration_System#ra2yr

Remember RA2 Spy, in TS they also Chameleon Spy that have few logic related to red alert 2, I suggest we expanded the TS Agent= logic to look like RA2 one and fix any bug as possible.

**Comment by JoyfulShush** [comment]
At the very least, I would like the AI to respect spies - currently it does not, forcing mappers to work around it with weird alliances.

**Reply by Rampastring** [reply]
In vanilla TS, AI entirely ignores disguised units.

CnCNet ts-patches specifically has a hack to make the AI see through the disguise so that Spies are not a no-brainer to use against the AI.

**Reply by JoyfulShush** [reply]
So we need a middle ground where mappers can at least control it (e.g. missions where you use a Spy to infiltrate a building), and a tag to control whether a unit (such as dogs) or even a team (such as patrol team with dogs) can notice the Spy.
